api/db-cdr/generate-cdr/make-bq-data.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+# This generates big query data that gets put in cloudsql
+# Counts and teh public and cdr data four cloudsql
+# note dev-up must be run to generate the schema
+# note run-local-data-migrations must be run to generate hard coded data from liquibase
+# note  the account must be authorized to perform gcloud and bq operations
+
+set -xeuo pipefail
+IFS=$'\n\t'
+
+
+# get options
+# --project=all-of-us-workbench-test *required
+
+# --cdr=cdr_version ... *optional
+
+USAGE="./generate-clousql-cdr/make-bq-data.sh --bq-project <PROJECT> --bq-dataset <DATASET> --workbench-project <PROJECT>"
+USAGE="$USAGE --account <ACCOUNT> --cdr-version=YYYYMMDD"
+while [ $# -gt 0 ]; do
+  echo "1 is $1"
+  case "$1" in
+    --account) ACCOUNT=$2; shift 2;;
+    --bq-project) BQ_PROJECT=$2; shift 2;;
+    --bq-dataset) BQ_DATASET=$2; shift 2;;
+    --workbench-project) WORKBENCH_PROJECT=$2; shift 2;;
+    --cdr-version) CDR_VERSION=$2; shift 2;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+if [ -z "${ACCOUNT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_DATASET}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${WORKBENCH_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${CDR_VERSION}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+
+#Check cdr_version is of form YYYYMMDD
+if [[ $CDR_VERSION =~ ^[0-9]{4}(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])$ ]]; then
+    echo "New CDR VERSION will be $CDR_VERSION"
+  else
+    echo "CDR Version doesn't match required format YYYYMMDD"
+    echo "Usage: $USAGE"
+    exit 1
+fi
+
+NEW_BQ_CDR_DATASET=cdr$CDR_VERSION
+schema_path=generate-cdr/bq-schemas
+
+# Check that bq_dataset exists and exit if not
+datasets=`bq --project=$BQ_PROJECT ls`
+if [ -z "$datasets" ]
+then
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+if [[ $datasets =~ .*$BQ_DATASET.* ]]; then
+  echo "$BQ_PROJECT.$BQ_DATASET exists. Good. Carrying on."
+else
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+
+
+# Make dataset for cdr cloudsql tables
+datasets=`bq --project=$WORKBENCH_PROJECT ls`
+echo $datasets
+if [[ $datasets =~ .*$NEW_BQ_CDR_DATASET.* ]]; then
+  echo "$NEW_BQ_CDR_DATASET exists"
```

Comment:
So if it already exists, it seems like we should maybe bail out? Or delete the whole thing and create a new dataset with the same name? I see that you remove tables below before recreating them, but if there was something else hanging out in the data set, not sure if we'd want to keep it...

Reply:
I tried that initially but I always get "dataset in use, can't delete" error. Thus deleting the tables ..   Bail out is not an option ...  I think this is fine.